### ISSUE TITLE
React-Dom version issue

### ISSUE THREAD

**Issue by Sangwon-sim-simacro** [adobe/react-spectrum-charts#34]
### Provide a general summary of the issue here

I used react, react-dom 18.2.0.
When I install and use react-spectrum-charts,
(I see, It has peerDependencies  react-dom "17.0.2 - latest")
Invalid hook error happens.

I think when I install adobe-spectrum-chart , the package use react-dom 17.0.2 only.
How can I solve this?

![image](https://github.com/adobe/react-spectrum-charts/assets/134468978/d27060be-f14b-46c4-a77f-4458cefeaadd)

### 🤔 Expected Behavior?

Chart successfully rendering when I use adobe/react-spectrum-charts

### 😯 Current Behavior

When I use adobe/react-specturm-charts, Invalid hook error happens.

### 💁 Possible Solution

When I install adobe/react-specturm-charts, 
If user using 18^ version react-dom, @adobe/react-specturm-chart pacakge should use 18 version react-dom.
becuase of predependence is 17.0.2 ~ latest.

### 🖥️ Steps to Reproduce

npx create-react-app test_app
npm install @adobe/react-spectrum-charts

just use
<Charts></Charts>
will cause invalide hooks error 

### Version

0.0.1.alpha.7

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

Windows

**Comment by marshallpete** [comment]
Updated how we define version ranges in this PR https://github.com/adobe/react-spectrum-charts/pull/35

Old method did not seem to work as expected

Also moved react and react-dom to devDependencies since they are already peerDependencies. This will prevent react-spectrum-charts from installing it's own version of react and react-dom

**Comment by marshallpete** [comment]
Fixed with `v0.0.1-alpha.8`

```
npx create-react-app test-app
cd test-app
npm i
npm i @adobe/react-spectrum-chart vega vega-lite
```

Replace `test-app/src/App.js` with:
```
import logo from "./logo.svg";
import "./App.css";
import { Axis, Bar, Chart } from "@adobe/react-spectrum-charts";

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
        <div style={{ width: "100%" }}>
          <Chart
            backgroundColor="#282c34"
            colorScheme="dark"
            maxWidth={800}
            data={[
              { x: 0, y: 0.5 },
              { x: 1, y: 1 },
              { x: 2, y: 2 },
              { x: 3, y: 1.5 },
            ]}
          >
            <Axis position="bottom" />
            <Axis position="left" />
            <Bar dimension="x" metric="y" />
          </Chart>
        </div>
      </header>
    </div>
  );
}

export default App;

```

run
```
npm start
```

**Comment by marshallpete** [comment]
Fixed with `v0.0.1-alpha.8`

```
npx create-react-app test-app
cd test-app
npm i
npm i @adobe/react-spectrum-chart vega vega-lite
```

Replace `test-app/src/App.js` with:
```
import logo from "./logo.svg";
import "./App.css";
import { Axis, Bar, Chart } from "@adobe/react-spectrum-charts";

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
        <div style={{ width: "100%" }}>
          <Chart
            backgroundColor="#282c34"
            colorScheme="dark"
            maxWidth={800}
            data={[
              { x: 0, y: 0.5 },
              { x: 1, y: 1 },
              { x: 2, y: 2 },
              { x: 3, y: 1.5 },
            ]}
          >
            <Axis position="bottom" />
            <Axis position="left" />
            <Bar dimension="x" metric="y" />
          </Chart>
        </div>
      </header>
    </div>
  );
}

export default App;

```

run
```
npm start
```

**Comment by Sangwon-sim-simacro** [comment]
It works fine in my side.
Thanks for deploying fixed version immediately
